### PR TITLE
Usage comparison notification service

### PR DESCRIPTION
**Pull request by manvirheer**

### CODE REVIEW

File: app/src/main/java/com/example/screentimemanager/usageComparisonService/UsageNotificationWorker.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        // If userEmail is still null, return failure
+        if (userEmail == null) {
+            Log.d("UsageNotificationWorker", "Notification In the worker class, userEmail $userEmail is null. Returning failure")
+            return Result.failure()
+        }
+
+        val c = Calendar.getInstance()
+        val year = c.get(Calendar.YEAR)
+        val month = c.get(Calendar.MONTH)
```
Comment (salmanayazz) [inline app/src/main/java/com/example/screentimemanager/usageComparisonService/UsageNotificationWorker.kt:44]: calendar.get(Calendar.MONTH) is 0 based so when the month is 12 it'll return 11. Right it i believe its checking for the previous month. Theres a function in the Util class that you could also use instead.

File: app/src/main/java/com/example/screentimemanager/usageComparisonService/UsageComparisonManager.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        Log.d("UsageComparisonManager", "Notification lowestUsageUserEmail: $lowestUsageUserEmail and usageTime: $usageTime");
+        // Fetch the first name of the user with the lowest usage
+        val user = userFirebaseDao.getUser(lowestUsageUserEmail)
+        val firstName = user?.firstName ?: "User"
+        
+        // Convert usage time to hours and minutes
+        val hours = usageTime / (60 * 60 * 1000)
+        val minutes = (usageTime % (60 * 60 * 1000)) / (60 * 1000)
+        val finalMessage = if (hours > 0) {
+            "$firstName got the lowest usage time today by only using $hours hours and $minutes minutes."
```
Comment (salmanayazz) [inline app/src/main/java/com/example/screentimemanager/usageComparisonService/UsageComparisonManager.kt:87]: Could you make these changes?
* add the date
* have the first name & last name
* add a case for when the user them self has the lowest screen time. In this case make the message somewhere along the line "Congrats! you had the lowest screen time among your friends on x day"
* don't show the notification to the user if they have no friends

Thanks

**Review by salmanayazz — DISMISSED** [review]

**Comment by manvirheer** [comment]
@salmanayazz can you check the recent commit. I have done all the requested work apart from util thing, didnt had to use it.
Screenshot 2023-12-05 at 12 04 20 PM

Screenshot 2023-12-05 at 12 03 51 PM
github.com/salmanayazz/screen-time-manager/assets/130617760/9f394437-1f36-411e-8dd4-ace70b12e7f4">

**Comment by salmanayazz** [comment]
It seems to send the notification every time the app is restarted. Also i couldn't see anything that indicated the notification was sent every day, did you implement that part?

**Comment by manvirheer** [comment]
@salmanayazz What do you think now? There is shared pref for normal check to not send again in same day. And when they login, we set the notifications again because the notification set for initial person doesn't have any friend.

**Review by salmanayazz — APPROVED** [review]

**Pull request merged**